ql/src/java/org/apache/hadoop/hive/ql/exec/vector/wrapper/VectorHashKeyWrapperGeneral.java

```diff
@@ -348,8 +348,8 @@ public void copyKey(KeyWrapper oldWrapper) {
         }
       } else {
         System.arraycopy(byteLengths, 0, clone.byteLengths, 0, byteValues.length);
-        Arrays.fill(byteStarts, 0);
-        System.arraycopy(byteStarts, 0, clone.byteStarts, 0, byteValues.length);
+        Arrays.fill(clone.byteStarts, 0);
+        // System.arraycopy(byteStarts, 0, clone.byteStarts, 0, byteValues.length);
```

Comment:
Please remove the line instead of commenting out.

Reply:
Fixed and added a test for this. Thanks very much for the review!